WordPress/openverse
# Add more docs for Plausible and auto-initialise custom event names

## Fixes




Fixes #1120 by @sarayourfriend
Fixes https://github.com/WordPress/openverse/issues/1119 by @sarayourfriend

## Description



Updates the existing analytics documentation with more comprehensive information about the local Plausible setup. I've also added documentation on how to work with custom events and a new script to automatically populate the local Plausible instance with the custom events so that we don't have to manually add them when testing.

## Testing Instructions


Read the updated documentation and confirm it makes sense and is accurate.

Test out the changes to `setup_plausible.sh` and confirm that the custom events get added as expected. Currently, there is only a single custom event defined, so to test that the loop logic works, add a test event to the `Events` type and re-run `just frontend/init`.

## Checklist



      `Update index.md`).
      a parent feature branch.
- [N/A] I added or updated tests for the changes I made (if applicable).
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
